languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -185,28 +359,95 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
   edge member -> def
 }
 
+(method_declaration (formal_parameters (_) @param)) @method
+{
+  edge @param.lexical_scope -> @method.lexical_scope
+  edge @method.lexical_scope -> @param.def
+  edge @method.defs -> @param.def
+}
+
+(formal_parameter type: (_) @type (_) @id) @param
+{
+  node @param.def
+  node @param.lexical_scope
+  node def__typeof
+
+  edge @type.lexical_scope -> @param.lexical_scope
+
+  attr (@param.def) node_definition = @id
+
+  attr (def__typeof) pop_symbol = ":"
+  edge def__typeof -> @type.type
+
+  edge @param.def -> def__typeof
+}
+
+(formal_parameter (modifiers) @modifiers) @this {
+  edge @modifiers.lexical_scope -> @this.lexical_scope
+}
+
+(method_declaration
+  (modifiers) @modifiers) @this {
+  edge @modifiers.lexical_scope -> @this.lexical_scope
+}
+
 (method_declaration
   body: (_) @stmt) @method
 {
   edge @stmt.before_scope -> @method.lexical_scope
 }
 
+(record_declaration name: (_) @name body: (_) @body) @this {
+  node def
+  attr (def) node_definition = @name
+  edge @this.defs -> def
+
+  edge @body.lexical_scope -> @this.lexical_scope
+}
+
+(record_declaration parameters: (formal_parameters (_) @param)) @this {
+  edge @param.lexical_scope -> @this.lexical_scope
+  edge @this.lexical_scope -> @param.def
+  edge @this.defs -> @param.def
+}
 
 ;; ============
 ;;  Statements
 ;; ============
 
-(expression_statement (_) @expr) @expr_stmt
+[
+  (assert_statement)
+  (block)
+  (break_statement)
+  (continue_statement)
+  (declaration)
+  (do_statement)
+  (expression_statement)
+  (enhanced_for_statement)
+  (for_statement)
+  (if_statement)
+  (labeled_statement)
+  (local_variable_declaration)
+  (return_statement)
+  (switch_expression)
+  (synchronized_statement)
+  (throw_statement)
+  (try_statement)
+  (try_with_resources_statement)
+  (while_statement)
+  (yield_statement)
+] @stmt
 {
-  node @expr_stmt.before_scope
-  node @expr_stmt.after_scope
-  edge @expr.lexical_scope -> @expr_stmt.before_scope
-  edge @expr_stmt.after_scope -> @expr_stmt.before_scope
+  node @stmt.before_scope
+  node @stmt.after_scope
```

Comment:
Simple chaining. Local variables can be declared at any point and affect the scope of later lines, so a single `lexical_scope` node would not suffice.